File: 商城后台接口开发/project_code/xlmall-api/src/main/java/com/whoiszxl/service/CategoryService.java
package com.whoiszxl.service;

import java.util.List;

import com.whoiszxl.common.ServerResponse;
import com.whoiszxl.entity.Category;

/**
 * 
 * @author whoiszxl
 *
 */
public interface CategoryService {
	ServerResponse<String> addCategory(String categoryName, Integer parentId);
    ServerResponse<String> updateCategoryName(Integer categoryId,String categoryName);
    ServerResponse<List<Category>> getChildrenParallelCategory(Integer categoryId);
    ServerResponse<List<Integer>> selectCategoryAndChildrenById(Integer categoryId);	
}	

File: kotlinAndroidNativeApp/mcode/GoodsCenter/src/main/java/com/whoiszxl/goods/service/CategoryService.kt
package com.whoiszxl.goods.service

import com.whoiszxl.goods.data.protocol.Category
import rx.Observable

/**
 * 商品分类业务层接口
 */
interface CategoryService {

    /*
        获取分类
     */
    fun getCategory(parentId:Int): Observable<MutableList<Category>?>
}
